diff -up gpsd-3.25/SConscript.sconsflags gpsd-3.25/SConscript
--- gpsd-3.25/SConscript.sconsflags	2024-04-22 15:14:25.672775265 +0200
+++ gpsd-3.25/SConscript	2024-04-22 15:18:33.935960954 +0200
@@ -619,12 +619,15 @@ for i in ["ARFLAGS",
           "SHLINKFLAGS",
           ]:
     if i in os.environ:
-        env.MergeFlags(Split(os.getenv(i)))
+        t = i
+        if t == "LDFLAGS":
+            t = "LINKFLAGS"
+        env.MergeFlags({t: Split(os.getenv(i))})
 
 
-# Keep scan-build options in the environment
+# Keep scan-build and rpm options in the environment
 for key, value in os.environ.items():
-    if key.startswith('CCC_'):
+    if key.startswith('CCC_') or key.startswith('RPM_'):
         env.Append(ENV={key: value})
 
 # Placeholder so we can kluge together something like VPATH builds.
